app/controllers/posts_controller.rb

```diff
@@ -1,14 +1,13 @@
 class PostsController < ApplicationController
   def index
-    @user = User.find(params[:user_id]) # Find the user by user_id
-    @posts = @user.posts
+    @user = User.find(params[:user_id])
+    @posts = @user.posts.paginate(page: params[:page], per_page: 5)
   end
```

Comment:
- [ ] Kindly make sure to solve the N+1 problem. To fix the issue you can use `includes` like the following to include the comments with the posts.

```ruby
@posts = @user.posts.includes(:comments)
```